app/models/upload_certificate_event.rb

```diff
@@ -1,6 +1,6 @@
 class UploadCertificateEvent < Event
   belongs_to_aggregate :certificate
-  data_attributes :value, :usage
+  data_attributes :value, :usage, :component_id
```

Comment:
We need to add a presence and persistence check for the component somewhere here.

I think you should do this by adding an reader and writer called `component` where you can pass the component as a object.  When `component=` is called you can set `self.component_id=component.id`. 

You can then create the event like this: `UploadCertificateEvent.create(usage: 'signing', value: good_cert_1, component: component)`

I did this in a very generic way during the spike [here](https://github.com/chrisholmes/event-sourcing-spike/blob/master/app/models/event.rb#L55)

Does that make sense?

Reply:
If you want to do that in a follow up PR I'm happy with that.